api/src/lib.rs

```diff
@@ -2600,6 +2600,126 @@ ffi_export!(
     }
 });
 
+ffi_export!(
+    fn Java_com_horizen_commitmenttree_CommitmentTree_nativeGetFwtLeaves(
+    _env: JNIEnv,
+    _commitment_tree: JObject,
+    _sc_id: jbyteArray
+) -> jobject
+{
+    let sc_id = {
+        let sc_id_bytes = parse_jbyte_array_to_vec(&_env, &_sc_id, FIELD_SIZE);
+        FieldElement::deserialize(sc_id_bytes.as_slice()).expect("Can't parse the input sc_id_bytes into FieldElement")
+    };
+
+    let commitment_tree = {
+
+        let t =_env.get_field(_commitment_tree, "commitmentTreePointer", "J")
+            .expect("Should be able to get field commitmentTreePointer");
+
+        read_mut_raw_pointer(&_env, t.j().unwrap() as *mut CommitmentTree)
+    };
+
+    match commitment_tree.get_fwt_leaves(&sc_id) {
+        Some(leaves) => {
+            let field_class =  _env.find_class("com/horizen/librustsidechains/FieldElement")
+                .expect("Should be able to find FieldElement class");
+
+            let initial_element = _env.new_object(field_class, "(J)V", &[
+                JValue::Long(0)]).expect("Should be able to create new long for FieldElement");
+
+            let leaf_fe_array = _env.new_object_array(leaves.len() as i32, field_class, initial_element)
+                .expect("Should be able to create array of FieldElements");
+
+            for (idx, leaf) in leaves.iter().enumerate() {
+                let leaf_field_ptr: jlong = jlong::from(Box::into_raw(Box::new(leaf.clone())) as i64);
+
+                let leaf_element = _env.new_object(field_class, "(J)V", &[
+                    JValue::Long(leaf_field_ptr)]).expect("Should be able to create new long for FieldElement");
+
+                _env.set_object_array_element(leaf_fe_array, idx as i32, leaf_element)
+                    .expect("Should be able to add FieldElement leaf to an array");
+            }
+
+
+            let cls_optional = _env.find_class("java/util/Optional").unwrap();
+
+            let empty_res = _env.call_static_method(cls_optional, "of", "(Ljava/lang/Object;)Ljava/util/Optional;", &[JValue::from(JObject::from(leaf_fe_array))])
+                .expect("Should be able to create new value for Optional");
+
+            *empty_res.l().unwrap()
```

Comment:
Again, not important, to save a little bit of code we might consider writing and using a generic function that wraps a JObject into another Optional JObject

Reply:
Thanks!
I completely agree with you. In these methods there is a lot of boilerplate that should be moved to common utility methods.
Currently, just reused the same approach as we have in other places.